ttl/cache/infoschema.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/parser/model"
+	"github.com/pingcap/tidb/sessionctx"
+)
+
+// InfoSchemaCache is the cache for InfoSchema, it builds a map from physical table id to physical table information
+type InfoSchemaCache struct {
+	baseCache
+
+	schemaVer int64
+	Tables    map[int64]*PhysicalTable
+}
+
+// NewInfoSchemaCache creates the cache for info schema
+func NewInfoSchemaCache(updateInterval time.Duration) *InfoSchemaCache {
+	return &InfoSchemaCache{
+		baseCache: newBaseCache(updateInterval),
+	}
+}
+
+// Update updates the info schema cache
+func (isc *InfoSchemaCache) Update(sctx sessionctx.Context) error {
+	is, ok := sctx.GetDomainInfoSchema().(infoschema.InfoSchema)
+	if !ok {
+		return errors.New("fail to get domain info schema from session")
+	}
+
+	ext, ok := is.(*infoschema.SessionExtendedInfoSchema)
+	if !ok {
+		return errors.New("fail to get extended info schema")
+	}
+
+	if isc.schemaVer == ext.SchemaMetaVersion() {
+		return nil
+	}
+
+	newTables := make(map[int64]*PhysicalTable, len(isc.Tables))
+	for _, db := range is.AllSchemas() {
+		for _, tbl := range is.SchemaTables(db.Name) {
+			tblInfo := tbl.Meta()
+			if !isTTLTable(tblInfo) {
+				continue
+			}
+
+			if tblInfo.Partition == nil {
+				ttlTable, err := isc.newTable(db.Name, tblInfo, nil)
```

Comment:
I'm thinking whether to just log the error and continues to load next table so that a broken table will not break the whole load.  A "broken table" maybe a bug, or some corner cases we did not consider, for example the middle state in schema change.

Reply:
Added. I'll also need some rethink about the availability of cache in https://github.com/pingcap/tidb/pull/39472, so I made it a draft temporarily.